### PR TITLE
2.1 to master

### PR DESCRIPTION
**Pull request by joonsp**

### CODE REVIEW

File: Event/Subscriber/ElasticaQuerySubscriber.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Knp\Component\Pager\Event\Subscriber\Paginate\ElasticaQuerySubscriber as BaseSubscriber;
+
+/**
+ * Elastica query pagination.
+ * Return the Elastica_ResultSet instead of it's results
+ */
+class ElasticaQuerySubscriber extends BaseSubscriber
+{
+    public function items(ItemsEvent $event)
```
Comment (janimatti) [inline Event/Subscriber/ElasticaQuerySubscriber.php:18]: Missing docblock

File: Form/ChoosableSearchType.php [hunk 2]
```diff
@@ -16,12 +16,13 @@ public function __construct($choices)
```
Comment (janimatti) [inline Form/ChoosableSearchType.php:18]: missing docblock

File: Form/ChoosableSearchType.php [hunk 3]
```diff
@@ -16,12 +16,13 @@ public function __construct($choices)
         $this->choices = $choices;
     }
 
-    public function buildForm(FormBuilder $builder, array $options)
+    public function buildForm(FormBuilderInterface $builder, array $options)
```
Comment (janimatti) [inline Form/ChoosableSearchType.php:26]: missing docblock

File: Form/SearchType.php [hunk 4]
```diff
@@ -3,17 +3,19 @@
...
 use Symfony\Component\Form\AbstractType;
-use Symfony\Component\Form\FormBuilder;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
 
 class SearchType extends AbstractType
 {
     
-    public function buildForm(FormBuilder $builder, array $options)
+    public function buildForm(FormBuilderInterface $builder, array $options)
```
Comment (janimatti) [inline Form/SearchType.php:16]: missing docblock

Comment (janimatti) [inline Form/SearchType.php:16]: missing docblock

File: Service/Search/FOQElasticaSearch.php [hunk 5]
```diff
@@ -35,16 +40,40 @@ public function __construct(Container $container)
...
+
         return $this->convertToSearchResult($elasticaResultSet);
     }
-    
+
+    /**
+     * Gets a paginator wrapping the result of a search
+     *
+     * @param  string $index
+     * @param  strgin $term
```
Comment (janimatti) [inline Service/Search/FOQElasticaSearch.php]: typo

File: Service/Search/FOQElasticaSearch.php [hunk 6]
```diff
@@ -56,22 +85,84 @@ protected function convertToSearchResult(\Elastica_ResultSet $elasticaResultSet)
                 $elasticaResult->getSource()
             );
         }
-        return new DefaultSearchResultSet($results, $elasticaResultSet->getTotalHits(), $elasticaResultSet->getTotalTime());
+
+        // disable call to getTotalTime until that functions exists in an ruflin/elastica version supported by elastica-bundle
+        // return new DefaultSearchResultSet($results, $elasticaResultSet->getTotalHits(), $elasticaResultSet->getTotalTime());
```
Comment (janimatti) [inline Service/Search/FOQElasticaSearch.php:90]: remove commented code

File: Service/Search/FOQElasticaSearch.php [hunk 7]
```diff
@@ -56,22 +85,84 @@ protected function convertToSearchResult(\Elastica_ResultSet $elasticaResultSet)
...
+        $mapFinder = $this->getFinder($index);
+
+        return $mapFinder->find($term, $limit);
+    }
+
+    /**
+     * Gets a paginator wrapping the result of a search
+     *
+     * @param  string $index
+     * @param  strgin $term
```
Comment (janimatti) [inline Service/Search/FOQElasticaSearch.php]: typo

File: Service/Search/FOQElasticaSearch.php (hunk 6 above)
Comment (joonsp) [inline Service/Search/FOQElasticaSearch.php:90]: this is here to remind that when a third part library supports this functionality, use it

**Pull request merged**